### DISCUSSION TITLE
Not cleared on the purpose of the mod

### DISCUSSION THREAD

**Discussion by lockf** [Sinytra/ForgifiedFabricAPI#5] [Q&A] [answered]
Does this mod allows the use of any fabric mods in forge or only the ones that use this API?

-lockf

**Comment by Su5eD** [comment, marked as answer]
Hi, the Forgified Fabric API is solely a port of the Fabric API, and does not contain any loader code that would allow loading Fabric mods on Forge. This will be handled by another of our (or more like mine heh) upcoming projects, Connector. More info on that soon!

**Reply by Rensik** [reply]
Isn't this useless until then, or is there a way to apply Fabric over Forge in such a way that this would then work?

**Reply by Su5eD** [reply]
As the readme explains, the FFAPI on its own serves the purpose of being a **cross-platform mod development library**, similar to existing ones such as Architectury or Balm. The difference is that rather than abstracting over multiple platforms, we apply the existing Fabric API on forge, which saves us time and effort maintaining it for two platforms at once. It is a very thin layer and does not cover any loader code, but it can make mod porting from fabric to forge easier for devs who are already familiar with Fabric's API.

> It is designed to make cross platform mod development easier by allowing developers to use Fabric API as a common library to interact with the game's code on both platforms. However, it is not an abstraction layer, and loader-specific code still needs to be handled separately for each platform.

In the future, I might explore ways to make [ForgifiedFabricLoader](https://github.com/Sinytra/ForgifiedFabricLoader) independent and run alongside FFAPI to provide common access to mod metadata, saving even more lines of platform-dependent code.

**Reply by Rensik** [reply]
I really do wish you luck in your endeavors. The modding community seriously needs it. We need everyone back under one platform again.

**Comment by Unitei-Dev** [comment]
Hello, as I understand it, this is just a library not for running mods with Fabric?

**Reply by Rensik** [reply]
right, we're waiting for his other mod, Connector

**Reply by Unitei-Dev** [reply]
How soon?

**Reply by Rensik** [reply]
when it gets released. no sooner
